drv/stm32h7-rng/src/main.rs

```diff
@@ -228,11 +233,42 @@ impl NotificationHandler for Stm32h7RngServer {
     }
 }
 
+#[cfg(feature = "ereport")]
+fn generate_restart_id(rng: &mut Stm32h7Rng) {
+    use task_packrat_api::Packrat;
+    const BYTES: usize = 16;
+
+    let mut bytes = [0u8; BYTES];
+    for i in 0..(BYTES / 4) {
+        'retry: loop {
+            match rng.read() {
+                Ok(word) => {
+                    let word_bytes = &word.to_ne_bytes();
+                    bytes[i * 4..i * 4 + 4].copy_from_slice(word_bytes);
+                    break 'retry;
+                }
+                Err(_) => {
+                    // XXX(eliza): what do we do if this fails?
+                    let _ = rng.attempt_recovery();
+                }
+            }
+        }
+    }
+
+    let id = u128::from_ne_bytes(bytes);
+    let packrat = Packrat::from(PACKRAT.get_task_id());
+    let result = packrat.set_ereport_restart_id(id);
+    ringbuf_entry!(Trace::SetRestartId(result));
+}
+
 #[export_name = "main"]
 fn main() -> ! {
     let mut rng = Stm32h7Rng::new();
     rng.init();
 
+    #[cfg(feature = "ereport")]
+    generate_restart_id(&mut rng);
```

Comment:
only doing this once when the RNG driver starts up means that if packrat crashes, it will never get a new restart ID. do we care about this? IIUC, other things that cache data in packrat don't have any mechanism for recovering from packrat faulting. if we wanted to have this recover from packrat death, we could have a notification to the RNG driver to say "im dead, please send me a new restart ID."

Reply:
A terrible option for resiliency would be for the RNG driver to call `generate_restart_id` once every _N_ seconds!